File: InjuryPreventionRecoveryTracker/src/InjuryPreventionRecoveryTracker.Core/Model/InjuryAggregate/Enums/InjuryType.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace InjuryPreventionRecoveryTracker.Core;

/// <summary>
/// Represents the type of injury.
/// </summary>
public enum InjuryType
{
    /// <summary>
    /// Muscle strain or pull.
    /// </summary>
    Strain = 0,

    /// <summary>
    /// Ligament sprain.
    /// </summary>
    Sprain = 1,

    /// <summary>
    /// Bone fracture.
    /// </summary>
    Fracture = 2,

    /// <summary>
    /// Tendon injury.
    /// </summary>
    Tendonitis = 3,

    /// <summary>
    /// Cartilage damage.
    /// </summary>
    CartilageDamage = 4,

    /// <summary>
    /// Overuse injury.
    /// </summary>
    Overuse = 5,

    /// <summary>
    /// Other or custom injury type.
    /// </summary>
    Other = 6,
}

File: InjuryPreventionRecoveryTracker/src/InjuryPreventionRecoveryTracker.WebApp/projects/injury-prevention-recovery-tracker/src/app/models/injury-type.ts
export enum InjuryType {
  Strain = 0,
  Sprain = 1,
  Fracture = 2,
  Tendonitis = 3,
  CartilageDamage = 4,
  Overuse = 5,
  Other = 6
}

export const INJURY_TYPE_LABELS: Record<InjuryType, string> = {
  [InjuryType.Strain]: 'Strain',
  [InjuryType.Sprain]: 'Sprain',
  [InjuryType.Fracture]: 'Fracture',
  [InjuryType.Tendonitis]: 'Tendonitis',
  [InjuryType.CartilageDamage]: 'Cartilage Damage',
  [InjuryType.Overuse]: 'Overuse',
  [InjuryType.Other]: 'Other'
};
